tsdb/shard.go

```diff
@@ -283,20 +528,18 @@ func (s *Shard) deleteMeasurement(name string, seriesKeys []string) error {
 			if err := b.Delete([]byte(k)); err != nil {
 				return err
 			}
-			if err := tx.DeleteBucket([]byte(k)); err != nil {
+			if err := tx.DeleteBucket([]byte(k)); err != nil && err != bolt.ErrBucketNotFound {
 				return err
 			}
+			delete(s.cache, k)
 		}
 
 		return nil
 	}); err != nil {
-		_ = s.Close()
```

Comment:
Yes, I don't think we should be closing the shard simply because a write failed. There's a lot of reasons why it could fail that could be temporary (e.g. `disk full`). If there's a legitimate fatal error with the write then the whole box is probably having issues.